### ISSUE TITLE
Should adding Tasks with the same ID work?

### ISSUE THREAD

**Issue by yeldiRium** [thenativeweb/marble-run#1]
Hi!

When reading the test cases and playing around with the library I noticed that the following works:

```js
const Course = require('marble-run');
const delay = require('delay');

const course = new Course();

console.log(Date.now());
const addTasks = Promise.all([
    course.add({
      routingKey: '000000-0000-0000-0000-0000000000001',
      id: '000000-0000-0000-0000-000000000000A',
      async task () {
        await delay(200);
        console.log(Date.now());
      }
    }),
    course.add({
      routingKey: '000000-0000-0000-0000-0000000000002',
      id: '000000-0000-0000-0000-000000000000A',
      async task () {
        await delay(100);
        console.log(Date.now());
      }
    })
  ]);
```

It results in the following output:

```
1557940094557
1557940094663
1557940094763
```

Meaning there are two tasks with the same ID being run at the same time. Should this be possible?
It will probably do no harm but could throw an error to prevent users of the library from accidentally adding a task twice.

Kind regards,
yeldiR

**Comment by goloroden** [comment]
Thanks for bringing this up 😊

TBH I think we didn‘t have this in mind when writing this module. Supposed there should be an error, the logic would probably be to check whether *currently* there is another task with the same ID, i.e. we would not keep track of past tasks that have already been completed. Is that how you‘d expect things to happen?

**Comment by yeldiRium** [comment]
> we would not keep track of past tasks
You're right. Additionally tracking all previous tasks is beyond the scope of this library, especially since it would need to persist those across process restarts in order to work long term (depending on use case).

Also since the ids are (at least in the tests) uuids it is very unlikely for the same id to appear multiple times over a longer stretch of time. My main intention was to prevent the accidental queueing of the same task multiple times which checking for currently running tasks should solve.

**Comment by yeldiRium** [comment]
I had a few minutes and opened a pull request with a test for it. Please let me know if I should change anything.
However no implementation yet :)

**Comment by goloroden** [comment]
Closed by #2
